src/driver/hermes_cdev.c

```diff
@@ -247,14 +246,20 @@ static ssize_t hermes_read_write_iter(struct kiocb *iocb, struct iov_iter *to)
 
 	pos = cfg->ehdsoff + offset + env->data_slot * cfg->ehdssze;
 	iov_iter_truncate(to, cfg->ehdssze - offset);
-	res = xdma_channel_read_write(chnl, to, pos);
+	ret = xdma_channel_read_write(chnl, to, pos);
 
 	if (write)
 		xdma_release_h2c(chnl);
 	else
 		xdma_release_c2h(chnl);
 
-	return res;
+	if (write && iocb->ki_flags & IOCB_SYNC) {
```

Comment:
If xdma_channel_read_write() fails, we probably should not do the fsync()....

Reply:
Ops, nice catch